datalad_webapp/tests/test_procedure.py

```diff
@@ -0,0 +1,89 @@
+import os
+import pytest
+import flask
+
+from datalad.api import create
+from datalad.api import webapp
+
+
+@pytest.fixture
+def client(tmpdir):
+    ds = create(tmpdir.strpath)
+
+    with open(os.path.join(ds.path, 'dummy_procedure.py'), 'w') as f:
+        f.write("import sys; print(\"This is a dummy procedure running in %s\" "
+                "% sys.argv[1]")
+    ds.add('dummy_procedure.py', to_git=True, save=False)
+
+    ds.config.add(
+        'datalad.locations.dataset-procedures',
+        '.',
+        where='dataset')
+    ds.config.add(
+        'datalad.procedures.dummy_procedure.call-format',
+        'python "{script}" "{ds}" nonsense argument',
+        where='dataset'
+    )
+    ds.config.add(
+        'datalad.procedures.dummy_procedure.help',
+        "This is a help message",
+        where='dataset'
+    )
+    ds.save(message="dummy procedure added")
+
+    res = webapp(
+        dataset=ds.path,
+        mode='dry-run',
+        return_type='item-or-list',
+    )
+    app = res['app']
+
+    client = app.test_client()
+
+    yield client, ds
+
+
+def test_server_startup(client):
+    client, ds = client
+    with client as c:
+        # unauthorized access is prevented
+        rv = client.get('/api/v1/procedure')
+        assert rv.status_code == 401
+        assert 'results' not in rv.get_json()
+        # we get no magic authentication
+        assert 'api_key' not in flask.session
+
+        # authenticate
+        rv = client.get('/api/v1/auth')
+        assert rv.status_code == 200
+
+        # request list of files
+        rv = client.get('/api/v1/procedure')
+        assert rv.status_code == 200
+
```

Comment:
Everything until here is not actually the scope of the test. Maybe we should move this into a common helper.